components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -15,25 +15,36 @@ import SearchIcon from 'components/widgets/icons/search_icon';
 
 import ChannelRow from './channel_row';
 
-export default class ChannelList extends React.PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            searchAllChannels: PropTypes.func.isRequired,
-            getData: PropTypes.func.isRequired,
-        }).isRequired,
-        data: PropTypes.array,
-        total: PropTypes.number,
+interface ChannelListProps {
+    actions: {
+        searchAllChannels: (term: string, notAssociatedToGroup?: string, excludeDefaultChannels?: boolean, page?: number, perPage?: number) => any;
+        getData: (page: number, perPage: number, notAssociatedToGroup? : string, excludeDefaultChannels?: boolean) => any;
+        removeGroup?: () => void;
     };
+    data?: {id: string; display_name: string}[];
+    total?: number;
+    removeGroup?: () => void;
+    onPageChangedCallback?: () => void;
+    emptyListTextId: string;
+    emptyListTextDefaultMessage: string;
+}
 
-    constructor(props) {
-        super(props);
-        this.state = {
-            searchString: '',
-            channels: [],
-            searchTotalCount: 0,
-            pageResetKey: 0,
-        };
-    }
+interface ChannelListState {
+    searchString: string;
+    channels: undefined[];
```

Comment:
undefined?

Reply:
@reflog  for channels. right now the channel and channel[] type is type any. Should I create a Ichannel interface in a separate file and use that? Should I take it all the fields from mm-server channel modal?

Reply:
there's Channel definition in mattermost-redux and it should be imported from there